wallets/provider-phantom/src/legacy/helpers.ts

```diff
@@ -0,0 +1,57 @@
+import type { Network } from '@rango-dev/wallets-core/legacy';
+import type { ProviderApi } from '@rango-dev/wallets-core/namespaces/evm';
+
+import { Networks } from '@rango-dev/wallets-core/legacy';
+
+type Provider = Map<Network, any>;
+
+export function phantom(): Provider | null {
+  const { phantom } = window;
+
+  if (!phantom) {
+    return null;
+  }
+
+  const { solana, ethereum } = phantom;
+
+  const instances: Provider = new Map();
+
+  if (ethereum && ethereum.isPhantom) {
+    instances.set(Networks.ETHEREUM, ethereum);
+  }
+
+  if (solana && solana.isPhantom) {
+    instances.set(Networks.SOLANA, solana);
+  }
+
+  return instances;
+}
+
+export function evmPhantom() {
+  const instances = phantom();
+
+  const evmInstance = instances?.get(Networks.ETHEREUM);
```

Comment:
You also used these functions in new provider.

Reply:
I moved them to `src/utils.ts` since they will be used in Hub even `legacy` being removed.

Reply:
I tried to make `phantom`, `evmPhantom` and `solanaPhantom` into one function but it's not possible. `phantom()` is using to detect wether the wallet is installed or not and also it's used in `legacy` implementation. 

`evmPhantom` and `solanaPhantom` are using `phantom` to access to each namespace separately and throw an error if they are not available. this is because they are using inside hub's namespaces and they should throw error if they are not available. And also they are reusing in namespace, for example, `evmPhantom` is passing to `changeAccountSubscriber` and `connect` separetly.

By removing those function the `provider-phantom/namespaces/evm.ts` will look like this:

```ts

const [changeAccountSubscriber, changeAccountCleanup] =
  actions.changeAccountSubscriber(() => {
    const evmInstance = phantom()?.get(Networks.ETHEREUM) || undefined;
    if (!evmInstance) {
      throw new Error(
        'Are you sure Phantom injected and you have enabled evm correctly?'
      );
    }
    return evmInstance;
  });

const evm = new NamespaceBuilder<EvmActions>('evm', WALLET_ID)
  .action('init', () => {
    console.log('[phantom]init called from evm cb');
  })
  .action([
    ...actions.recommended,
    actions.connect(() => {
      const evmInstance = phantom()?.get(Networks.ETHEREUM) || undefined;
      if (!evmInstance) {
        throw new Error(
          'Are you sure Phantom injected and you have enabled evm correctly?'
        );
      }
      return evmInstance;
    }),
  ])
  ...
```